auth0/structure_auth0_connection.go

```diff
@@ -385,6 +386,13 @@ func flattenConnectionOptionsEmail(options *management.ConnectionOptionsEmail) (
 	}
 	m["upstream_params"] = upstreamParams
 
+	if options.AuthParams != nil {
+		v, ok := options.AuthParams.(map[string]string)
+		if ok {
```

Comment:
Do we want to throw an error if auth params are not a map? Or at least a warning (using the diagnostics pkg)?

Reply:
I think throwing an error is a little heavy-handed here, but certainly agree that some action is necessary. I think logging a warning strikes a good balance.

Reply:
Added the warning. However, I'm unable to get it to render in the console, despite that LoC being reached and appropriate log verbosity configured.

Reply:
To raise a warning you need to use a diagnostics message with SeverityWarning: https://github.com/hashicorp/terraform-plugin-sdk/blob/70ce77bce6118b74a49762bb401b46a723c0bab8/diag/diagnostic.go#L101. You can get inspiration from the hook resource: https://github.com/auth0/terraform-provider-auth0/blob/54bff9bcc4c4c949a7d9fa10425dd9bfdd29085b/auth0/resource_auth0_hook.go#L152

Reply:
This logging is performed in this upstream PR: #241